docs/docs/getting_started/tooling/noir_codegen.md

```diff
@@ -0,0 +1,135 @@
+---
+title: Noir Codegen for Typescript
+description: Learn how to use Noir codegen to generate Typescript bindings 
+keywords: [Nargo, Noir, compile, TypeScript]
+sidebar_position: 2
+---
+
+You can generate TypeScript bindings with `noir_codegen`, allowing for easier and faster integration of your Noir circuits into TypeScript projects.
```

Comment:
```suggestion
For most development flows, you will want want to know the outputs of your circuit in an easy, programmable way. Some devs use external libraries for this, which are often hard to find (ex. for `Poseidon2` hashes) or unmaintained (ex. `circomlibjs`).

To help you, `noir_codegen` turns Noir circuits into TypeScript projects. It takes whatever Noir function you have and turns it into a TS bind you can run and see its output.
```

Reply:
I'll update to include some of this additional context.